src/components/screens/vacation-requests-screen.tsx

```diff
@@ -52,9 +52,9 @@ const VacationRequestsScreen = () => {
    * Fetch vacation request statuses
    */
   const fetchVacationRequestStatuses = async () => {
+    setLoading(true);
```

Comment:
I don't think this is necessary, and typically it's a good idea for loading state to be started from within the try unless there is a reason not to (as in the case of the other places you have made this change).

You should still add the setLoading(false) after the catch as you have done here. Was there no setLoading(false) previously?